hack/update-example.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh -e
+#
+# Updates book/src/Example.md with the latest revisions of all referenced files.
+# While doing so it prints a diff between files and a commit log.
+#
+# Updating the links to files embedded in the docs repo itself together
+# with the Readme.md is a bit harder:
+# - commit the updated .yaml files locally
+# - manually update the revision hash
+# - submit the PR *without* touching the updated .yaml files
+#
+# It's easier to update the .yaml files, submit a PR, and then
+# once accepted, update the example using this script.
+
+example=book/src/Example.md
+
+for url in $(grep 'kubectl create -f.*https://raw.githubusercontent.com/' $example | sed -e 's;.*\(https://raw.githubusercontent.com[^ (]*\).*;\1;'); do
+    # split URL into pieces
+    eval $(echo $url | sed -e 's/\]$//' -e 's;https://raw.githubusercontent.com/\([^/]*/[^/]*\)/\([^/]*\)/\(.*\);repo=\1 rev=\2 file=\3;')
+    git fetch https://github.com/$repo refs/heads/master:refs/remotes/$repo/master
```

Comment:
Instead of latest, would it make more sense to pull specific tags from released versions?

Reply:
@msau42 That makes more sense in a lot of ways, my concern though would be making sure we *remember* to update things like the docs repo.  That being said, it would be easy to just set a version file and update it with each CSI release and update that way.  Could even potentially automate  it at some point?

Reply:
Yes, we could setup some fancy CI that automates a lot of this.  In the meantime, it will have to be one of those things to remember to do with every release.

Reply:
I had considered a parameter to pull from a branch, but then didn't add that because I wasn't sure whether we would ever need it. If needed and easier than manually updating the Example.md, it can still be added at that time.